mono/mini/interp/interp.c

```diff
@@ -2907,13 +2907,47 @@ static int opcode_counts[512];
 #define frame_objref(frame) o
 #endif
 
+// This function is outlined to help save stack in its caller, on the premise
+// that it is relatively rarely called. This also lets it use alloca.
+static
+MONO_NEVER_INLINE
+void
+mono_interp_calli_nat_dynamic_pinvoke (
+	// Parameters are sorted by name.
```

Comment:
:D

Reply:
In the NT kernel locals are sorted by name.
It makes it so you don't have to spend time thinking about it, uh, once you pick the names (possibly picking them so they sort how you like).
Here there were too many to come up with a sensible ordering.